subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -394,84 +394,83 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
     }
 
     @Issue("https://github.com/gradle/gradle/issues/9586")
-    def "change in case of input files and folders will still syncs properly"() {
+    def "change in case of input file will still sync properly"() {
         given:
+        def lowercaseFile = file('file.txt').createFile()
+        def uppercaseFile = new File(lowercaseFile.parentFile, 'FILE.TXT') // make sure case is kept
```

Comment:
I think this should be enough:

```suggestion
        def uppercaseFile = file('FILE.TXT')
```

Reply:
it should, but it won't work, because the `file.txt` already exists, and it will be created with lowercase name

Reply:
<img width="450" alt="Screen Shot 2019-06-13 at 10 50 49" src="https://user-images.githubusercontent.com/1841944/59418323-2201d880-8dc9-11e9-8dad-f9f7e946dd52.png">

Reply:
How about this?

```groovy
    def lowercaseFile = file('file.txt')
    def uppercaseFile = file('FILE.TXT')
    lowercaseFile.createFile()
```

Reply:
That could work, but do you think it's easier to understand?

Reply:
Done (changed their order)